### PR TITLE
Pyic 3295 secure the network access to the stub

### PR DESCRIPTION
**Pull request by TobySaundersGDS**
This adds a RESTAPI gateway to the CIMIT stub with a usage plan, and an API key

### What changed
Added an API key as a basic barrier to entry

### Why did it change
To prevent accidental discovery of the CIMIT Stub service 

### Other considerations
The API definition needs fleshing out - currently just defining /healthcheck.
If this needs to be called within the VPC we will need to define DNS for it as otherwise traffic is blackholed.

### CODE REVIEW

File: di-ipv-cimit-stub/deploy/template.yaml [hunk 1]
```diff
@@ -317,3 +339,50 @@ Resources:
...
+          Name: "AWS::Include"
+          Parameters:
+            Location: "../openAPI/cimit-external.yaml"
+      ApiKeySourceType: HEADER #pragma: allowlist secret
+      Auth:
+        ApiKeyRequired: true #pragma: allowlist secret
+        UsagePlan:
+          CreateUsagePlan: PER_API
+          UsagePlanName: "CIMIT Usage Plan"
+        #ResourcePolicy:
```
Comment (therealslimsladey) [inline di-ipv-cimit-stub/deploy/template.yaml]: Remove commented out config?

File: .cfnlintrc.yaml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+ignore_checks:
+  - W3005
```
Comment (therealslimsladey) [inline .cfnlintrc.yaml:2]: Can this be a lint config for cimit stub only?

**Review by therealslimsladey — DISMISSED** [review]
lgtm, minor comments only

File: .cfnlintrc.yaml (hunk 2 above)
Comment (TobySaundersGDS) [reply .cfnlintrc.yaml:2]: I decided to leave it as I expect to be doing a similar thing for the orch stub and maybe others (also just moving it to the template folders  didn't work)

File: di-ipv-cimit-stub/deploy/template.yaml (hunk 1 above)
Comment (TobySaundersGDS) [reply di-ipv-cimit-stub/deploy/template.yaml]: yes

**Review by lwebb-iig — APPROVED** [review]

**Pull request merged**